pkg/tfbridge/tokens/tokens.go

```diff
@@ -51,11 +52,43 @@ func MakeStandard(pkgName string) Make {
 		if name == "" {
 			return "", fmt.Errorf("missing name for module %q", module)
 		}
+
+		name = makeSafeTokenPart(name)
+
 		lowerName := string(unicode.ToLower(rune(name[0]))) + name[1:]
 		return fmt.Sprintf("%s:%s/%s:%s", pkgName, module, lowerName, name), nil
 	}
 }
 
+// makeSafeTokenPart makes part safe to use as a token segment.
+//
+// makeSafeTokenPart fixes:
+// - parts that start with numbers, since these are not valid in Pulumi schema.
+func makeSafeTokenPart(part string) string {
+	return startsWithNumeric.ReplaceAllStringFunc(part, func(s string) string {
+		w, ok := digitToWord[s[0:1]]
+		if !ok {
+			return s
```

Comment:
Should we add a fallback for any other things?

Reply:
Numbers are the only thing I can think of that would be legal in the middle of a TF token but illegal in the beginning of a Pulumi token. What else are you thinking?

Reply:
I think Mihail mentioned an azure-native resource with a *?

Reply:
But that might not be valid in TF, so probably fine.

Reply:
`*` is a valid enum value, but it isn't a valid TF path componenet.